core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -605,6 +603,29 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  def unZip(inFile: File, unzipDir: File): Unit = {
+    if (!unzipDir.mkdirs && !unzipDir.isDirectory) {
+      throw new IOException("Mkdirs failed to create " + unzipDir)
+    } else {
+      if (Shell.WINDOWS) {
+        FileUtil.unZip(inFile, unzipDir)
```

Comment:
We should think about the case when `unzip` is not installed in Linux. Is there any other ways to avoid to rely on external programs?

Reply:
Because the zip file is decompressed by creating a new file, the permission cannot be saved, but the metadata of the permission can be saved when the unzip command is executed on Linux, but I don't know what special processing is done by the unzip command in Linux. Moreover, I found many API attempts of zip, and found that they were similar to the unzip of JDK, and there was no way to save the permission information.

Reply:
I've seen many JDK bugs and found that the unzip decompression permission problem seems to have lasted for a long time, such as https://bugs.openjdk.java.net/browse/JDK-6194856. Using java APIs doesn't seem to solve this problem

Reply:
@HyukjinKwon @srowen  Hello, I have found a way to store permissions for files, but this requires injecting two dependency projects, is this ok?

Reply:
+1 agree with @HyukjinKwon